src/taxonomy/data/apiHooks.js

```diff
@@ -12,95 +12,178 @@
- * @returns {import("./types.mjs").TaxonomyListData | undefined}
+ * Use this mutation to import a new taxonomy.
  */
-export const useTaxonomyListDataResponse = (org) => {
-  const response = useTaxonomyListData(org);
-  if (response.status === 'success') {
-    return { ...response.data, refetch: response.refetch };
-  }
-  return undefined;
+export const useImportNewTaxonomy = () => {
+  const queryClient = useQueryClient();
+  return useMutation({
+    /**
+    * @type {import("@tanstack/react-query").MutateFunction<
+    *   import("./types.mjs").TaxonomyData,
+    *   any,
+    *   {
+    *     name: string,
+    *     exportId: string,
+    *     description: string,
+    *     file: File,
+    *   }
+    * >}
+    */
+    mutationFn: async ({
+      name, exportId, description, file,
+    }) => {
+      const formData = new FormData();
+      formData.append('taxonomy_name', name);
+      formData.append('taxonomy_export_id', exportId);
+      formData.append('taxonomy_description', description);
+      formData.append('file', file);
+
+      const { data } = await getAuthenticatedHttpClient().post(apiUrls.createTaxonomyFromImport(), formData);
+      return camelCaseObject(data);
+    },
+    onSuccess: (data) => {
+      // There's a new taxonomy, so the list of taxonomies needs to be refreshed:
+      queryClient.invalidateQueries({
+        queryKey: taxonomyQueryKeys.taxonomyList(),
+      });
+      queryClient.setQueryData(taxonomyQueryKeys.taxonomyMetadata(data.id), data);
+    },
+  });
 };
 
 /**
- * Returns the status of the taxonomy list query
- * @param {string} org Optional organization param
- * @returns {boolean}
+ * Build the mutation to import tags to an existing taxonomy
  */
-export const useIsTaxonomyListDataLoaded = (org) => (
-  useTaxonomyListData(org).status === 'success'
-);
+export const useImportTags = () => {
+  const queryClient = useQueryClient();
+  return useMutation({
+    /**
+    * @type {import("@tanstack/react-query").MutateFunction<
+    *   import("./types.mjs").TaxonomyData,
+    *   any,
+    *   {
+    *     taxonomyId: number,
+    *     file: File,
+    *   }
+    * >}
+    */
+    mutationFn: async ({ taxonomyId, file }) => {
+      const formData = new FormData();
+      formData.append('file', file);
 
-/**
- * @param {number} taxonomyId
- * @returns {Pick<import('@tanstack/react-query').UseQueryResult, "error" | "isError" | "isFetched" | "isSuccess">}
- */
-export const useTaxonomyDetailDataStatus = (taxonomyId) => {
-  const {
-    isError,
-    error,
-    isFetched,
-    isSuccess,
-  } = useTaxonomyDetailData(taxonomyId);
-  return {
-    isError,
-    error,
-    isFetched,
-    isSuccess,
-  };
+      try {
+        const { data } = await getAuthenticatedHttpClient().put(apiUrls.tagsImport(taxonomyId), formData);
+        return camelCaseObject(data);
+      } catch (/** @type {any} */ err) {
+        throw new Error(err.response?.data.error || err.message);
```

Comment:
The previous version of this (in `api.py`) had `err.response?.data` but I think that's wrong. I changed it to `err.response?.data.error`, which is the same as the "import plan" endpoint had.